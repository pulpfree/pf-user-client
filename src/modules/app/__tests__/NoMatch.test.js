/* eslint no-undef: "off" */
import React from 'react'

import { mount } from 'enzyme'

import NoMatch from '../components/NoMatch'

describe('NoMatch', () => {
  let props
  let mountedC
  const noMatchC = () => {
    if (!mountedC) {
      mountedC = mount(
        <NoMatch {...props} />
      )
    }
    return mountedC
  }

  beforeEach(() => {
    props = {
      location: {pathname: '/foo'},
    }
    mountedC = undefined
  })

  it('always renders a component', () => {
    expect(noMatchC().find('div').length).toBeGreaterThan(0)
  })

  it('matches snapshot', () => {
    expect(noMatchC()).toMatchSnapshot()
  })

})