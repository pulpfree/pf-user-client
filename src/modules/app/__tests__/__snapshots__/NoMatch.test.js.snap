// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoMatch matches snapshot 1`] = `
ReactWrapper {
  "length": 1,
  Symbol(enzyme.__unrendered__): <NoMatch
    location={
        Object {
            "pathname": "/foo",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "location": Object {
        "pathname": "/foo",
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": <div>
        <h3>
                No match for 
                <code>
                        /foo
                </code>
        </h3>
</div>,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <h3>
          No match for 
          <code>
                    /foo
          </code>
</h3>,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": <h3>
            No match for 
            <code>
                        /foo
            </code>
</h3>,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "No match for ",
              <code>
                /foo
</code>,
            ],
          },
          "ref": null,
          "rendered": Array [
            "No match for ",
            Object {
              "instance": <code>
                /foo
</code>,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "/foo",
              },
              "ref": null,
              "rendered": Array [
                "/foo",
              ],
              "type": "code",
            },
          ],
          "type": "h3",
        },
      ],
      "type": "div",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "location": Object {
          "pathname": "/foo",
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": <div>
          <h3>
                    No match for 
                    <code>
                              /foo
                    </code>
          </h3>
</div>,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <h3>
            No match for 
            <code>
                        /foo
            </code>
</h3>,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": <h3>
              No match for 
              <code>
                            /foo
              </code>
</h3>,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "No match for ",
                <code>
                  /foo
</code>,
              ],
            },
            "ref": null,
            "rendered": Array [
              "No match for ",
              Object {
                "instance": <code>
                  /foo
</code>,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "/foo",
                },
                "ref": null,
                "rendered": Array [
                  "/foo",
                ],
                "type": "code",
              },
            ],
            "type": "h3",
          },
        ],
        "type": "div",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
